src/rules/extensions.js

```diff
@@ -37,15 +39,15 @@ module.exports = function (context) {
     const extension = path.extname(resolvedPath || importPath).substring(1)
 
     if (!extension || !endsWith(importPath, extension)) {
-      if (isUseOfExtensionEnforced(extension)) {
+      if (isUseOfExtensionRequired(extension) && !isUseOfExtensionForbidden(extension)) {
```

Comment:
What are the reasons that `externals.hasOwnProperty(key)` could break? Is that just when externals is nil?
If you feel like it's better to have it, keep it, I'm not that against it, just didn't see the value if it was just to check if `externals[key]` exists.